Code/Framework/AzFramework/AzFramework/AzFrameworkNativeUIModule.cpp

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <AzFramework/AzFrameworkNativeUIModule.h>
+#include <AzCore/Debug/Budget.h>
+
+// Component includes
+#include <AzFramework/Components/NativeUISystemComponent.h>
+
+AZ_DEFINE_BUDGET(AzFrameworkNativeUI);
+
+namespace AzFramework
+{
+    AzFrameworkNativeUIModule::AzFrameworkNativeUIModule()
+        : AZ::Module()
+    {
+        m_descriptors.insert(m_descriptors.end(), {
+            AzFramework::NativeUISystemComponent::CreateDescriptor(),
+        });
+    }
```

Comment:
Is the `AzFramework::NativeUISystemComponent` not a required system component when the headless server is not active?
I don't see an override of `GetRequiredSystemComponents` for this module